src/SubModules/MatrixModule.jl

```diff
@@ -75,4 +81,501 @@ function writePixmap(filename::String,matrix::Matrix)
+	a::Array{Tuple{Int64,Int64},1}	= []
+	[push!(a,(pattern[2*(i-1)+1],pattern[2*(i-1)+2])) for i in 1:len]
+	return a
+end
+
+#TODO: findlast für Array von Tuples vom Typ findlast(pattern,(Int,3)), wuerde computeTuplePatternFromPattern sparen
+"""
+    multiplySpMM(SpM,M)
+returns the product M2 = SpM*M
+"""
+function multiplySpMM(SpM::BCSRSpM,M::Array{Float64,2})
+	if SpM.colpattern[end][2] != size(M,1) error("Dimensions do not match ԅ(≖‿≖ԅ)") end
+	s		= size(M,2)
+	pattern = collect(Iterators.flatten(SpM.rowpattern))
+	pattern = pattern[1:findlast(pattern,s)]
+	b		= computeTuplePatternFromPattern(pattern)
+	res		= zeros(SpM.rowpattern[end][2],size(M,2))	
+	row		= computeRowFromRowptr(SpM)	
+	
+	for i = 1:length(SpM.val)
+		for j = 1:length(b)
+			res[SpM.colpattern[row[i]][1]:SpM.colpattern[row[i]][2],b[j][1]:b[j][2]] += SpM.val[i]*M[SpM.colpattern[SpM.col[i]][1]:SpM.colpattern[SpM.col[i]][2],b[j][1]:b[j][2]]
+		end
+	end
+	
+	return res
+end
+
+"""
+    multiplySpMSpM(SpM1,SpM2)
+returns the product M = SpM1*SpM2
+"""
+function multiplySpMSpM(SpM1::BCSRSpM,SpM2::BCSRSpM)
+	if SpM1.colpattern != SpM2.rowpattern error("Patterns do not match (ﾉ◕ヮ◕)ﾉ*:・ﾟ✧") end
+	res		= zeros(SpM1.rowpattern[end][2],SpM2.colpattern[end][2])
+	row1	= computeRowFromRowptr(SpM1)
+	row2	= computeRowFromRowptr(SpM2)
+
+	for i = 1:length(SpM1.val), j = 1:length(SpM2.val)
+			if SpM1.col[i] == row2[j]
+				res[SpM1.colpattern[row1[i]][1]:SpM1.colpattern[row1[i]][2],SpM2.rowpattern[SpM2.col[j]][1]:SpM2.rowpattern[SpM2.col[j]][2]] += SpM1.val[i]*SpM2.val[j]
+			end
+	end
+	
+	return res
+end
+
+"""
+    multiplySpMSpM2(SpM1,SpM2)
+returns the product SpM3 = SpM1*SpM2
+"""
+function multiplySpMSpM2(SpM1::BCSRSpM,SpM2::BCSRSpM)
+	if SpM1.colpattern != SpM2.rowpattern error("Patterns do not match (ﾉ◕ヮ◕)ﾉ*:・ﾟ✧") end
+	row1 = computeRowFromRowptr(SpM1)
+	row2 = computeRowFromRowptr(SpM2)
+	
+	dict::Dict{Tuple{Int64,Int64},Array{Float64,2}} = Dict{Tuple{Int64,Int64},Array{Float64,2}}() 
+	
+	for i = 1:length(SpM1.val), j = 1:length(SpM2.val)
+			if SpM1.col[i] == row2[j]
+				if haskey(dict,(row1[i],SpM2.col[j])) 
+					dict[(row1[i],SpM2.col[j])] += SpM1.val[i]*SpM2.val[j]
+				else 
+					dict[(row1[i],SpM2.col[j])] = SpM1.val[i]*SpM2.val[j]
+				end
+			end
+	end
+	
+	dict2 = sort(collect(dict), by = x->(x[1],x[2]))
+	rowptr = computeRowptrFromRow([x[1][1] for x in dict2],SpM1)
+	col = [x[1][2] for x in dict2]
+	val = [x[2] for x in dict2]
+	
+	return BCSRSpM(val,col,rowptr,SpM1.colpattern,SpM2.rowpattern)
+end
+
+#TODO: push! geht nicht und col muss gedreht werden
+function computeMatrixTransposeOfSpM(SpM::BCSRSpM)
+	row = computeRowFromRowptr(SpM)
+	SpMT::BCSRSpM = BCSRSpM([],[],[],SpM.rowpattern,SpM.colpattern)
+	for i = 1:length(SpM.val)
+		push!(SpMT.val,SpM.val[i]')
+		push!(SpMT.col,row[i])
+		row[i]			= SpM.col[i]
+	end
+	SpMT.rowptr = computeRowptrFromRow(row,SpM2)
+	return SpMT
+end
+
+#@Henry: laesst sich so nicht in MatrixModule aufrufen weil ShellModule benutzt
```

Comment:
I'm not sure if benchmarking should be a part of QuantumLab itself. Maybe, if you want to, we could have a separate benchmark.jl script in QL/test to separate it from the code functionalities. Or you could keep that script separate (not commiting it to QuantumLab and just run it when you are ready and report the results in your lab report.